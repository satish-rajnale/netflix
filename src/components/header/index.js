import { useState } from "react";
import { Link as ReactRouterLink } from "react-router-dom";
import {
  Profile,
  PlayButton,
  Picture,
  Dropdown,
  Group,
  Link,
  FeatureCallOut,
  Text,
  Background,
  ButtonLink,
  Container,
  Logo,
  Feature,
  SearchInput,
  SearchIcon,
  Search,
} from "./styles/header";
import React from 'react';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature>{children}</Feature>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((searchActive) => !searchActive)}
      >
        <img src="/images/icons/search.png" alt="search" />
      </SearchIcon>
      <SearchInput
        placeholder="Search films & series"
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        active={searchActive}
      />
    </Search>
  );
};
