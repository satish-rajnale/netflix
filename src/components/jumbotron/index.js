import React from 'react';
import {Inner} from './styles/jumbotron'

export default function Jumbotron({children, direction = 'row', ...restProps}){
    return (
        <Inner direction={direction}>
            <p>Hello again</p>
        </Inner>
    )
}


Jumbotron.Container = function JumbotronContainer({ children, ...restProps}){
    return <Container {...restProps}>{children}</Container>
}