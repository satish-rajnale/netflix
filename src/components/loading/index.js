import { LockBody, Spinner, Picture, ReleaseBody } from "./styles/loading";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} />
    </Spinner>
  );
}

Loading.ReleaseBody = function LoadingReleasebody() {
  return <ReleaseBody />;
};
