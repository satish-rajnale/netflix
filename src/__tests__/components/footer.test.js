import React from 'react';
import { render, fireEvent, cleanup} from  "@testing-library/react";
import {Footer} from '../../components';


describe("<Footer/>", ()=> {
    afterEach(cleanup)
    it("renders the <Footer/> with data", ()=> {
        const {container, getByText, getAllByText}  = render(<Footer>
            <Footer.Title>Questions? Call 000-800-040-1843</Footer.Title>
            <Footer.Break />
            <Footer.Row>
              <Footer.Column>
                <Footer.Link href="#">FAQ</Footer.Link>
                <Footer.Link href="#">Investor Relations</Footer.Link>
                <Footer.Link href="#">Ways To Watch</Footer.Link>
                <Footer.Link href="#">Media Center</Footer.Link>
              </Footer.Column>
      
              <Footer.Column>
                <Footer.Link href="#">Help Centre</Footer.Link>
                <Footer.Link href="#">Investor Relations</Footer.Link>
                <Footer.Link href="#">Terms Of Use</Footer.Link>
                <Footer.Link href="#">Corporate Information</Footer.Link>
              </Footer.Column>
      
              <Footer.Column>
                <Footer.Link href="#">Account</Footer.Link>
                <Footer.Link href="#">Jobs</Footer.Link>
                <Footer.Link href="#">Privacy</Footer.Link>
                <Footer.Link href="#">Contact Us</Footer.Link>
              </Footer.Column>
      
              <Footer.Column>
                <Footer.Link href="#">Cookie Preferences</Footer.Link>
                <Footer.Link href="#">Netflix Originals</Footer.Link>
                <Footer.Link href="#">Legal Notices</Footer.Link>
                <Footer.Link href="#">Ways To Watch</Footer.Link>
              </Footer.Column>
            </Footer.Row>
            <Footer.Break />
            <Footer.Text>Netflix India</Footer.Text>
          </Footer>);


    expect(getByText("Questions? Call 000-800-040-1843")).toBeTruthy();
    expect(getByText("FAQ")).toBeTruthy();
    expect(getAllByText("Investor Relations")).toBeTruthy();
    expect(getAllByText("Ways To Watch")).toBeTruthy();
    expect(getByText("Media Center")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();


    })
})