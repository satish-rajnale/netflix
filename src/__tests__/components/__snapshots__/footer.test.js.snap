// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> renders the <Footer/> with data 1`] = `
<div
  class="footer__Container-qu6i6-0 kMAYhg"
>
  <p
    class="footer__Title-qu6i6-3 bKhpHO"
  >
    Questions? Call 000-800-040-1843
  </p>
  <div
    class="footer__Break-qu6i6-6 UrvGN"
  />
  <div
    class="footer__Row-qu6i6-1 dWrLbY"
  >
    <div
      class="footer__Column-qu6i6-2 jtLmfk"
    >
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Ways To Watch
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Media Center
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-2 jtLmfk"
    >
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Help Centre
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Terms Of Use
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Corporate Information
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-2 jtLmfk"
    >
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-2 jtLmfk"
    >
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Netflix Originals
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Legal Notices
      </a>
      <a
        class="footer__Link-qu6i6-4 ipqJdV"
        href="#"
      >
        Ways To Watch
      </a>
    </div>
  </div>
  <div
    class="footer__Break-qu6i6-6 UrvGN"
  />
  <div
    class="footer__Text-qu6i6-5 fDDZSy"
  >
    Netflix India
  </div>
</div>
`;
